go-controller/pkg/ovn/egressip.go

```diff
@@ -384,16 +382,23 @@ func (e *egressIPMode) getGatewayRouterJoinIP(node string, wantsIPv6 bool) (net.
 	return nil, fmt.Errorf("node does not have a gateway router IP corresponding to the IP version requested")
 }
 
-func (e *egressIPMode) getPodIPs(pod *kapi.Pod) ([]net.IP, error) {
-	_, podIps, err := util.GetPortAddresses(getPodKey(pod), e.ovnNBClient)
-	if err != nil || len(podIps) == 0 {
+func (e *egressIPMode) getPodIPs(pod *kapi.Pod) []net.IP {
+	if len(pod.Status.PodIPs) == 0 && pod.Status.PodIP == "" {
 		e.podRetry.Store(getPodKey(pod), true)
-		return nil, fmt.Errorf("pod does not have IP assigned, err: %v", err)
+		klog.V(5).Info("Pod does not have an IP assigned, retrying until it gets one")
```

Comment:
except this function doesn't do any retry, and in the delete case (eg, if a pod is created and then deleted before CNI succeeds and so it never gets a pod IP) it wouldn't be retried so the message is wrong. So don't log anything here. You can log it in the caller if you need a log message